File: split/printcs.pas
procedure printcs(p: integer);
begin
  if p < 514 then
    if p >= 257 then
      if p = 513 then
      begin
        printesc(504);
        printesc(505);
        printchar(32);
      end
      else
      begin
        printesc(p - 257);
        if eqtb[3983 + p - 257].hh.rh = 11 then
          printchar(32);
      end
    else if p < 1 then
      printesc(506)
    else
      print(p - 1)
  else if p >= 2881 then
    printesc(506)
  else if (hash[p].rh < 0) or (hash[p].rh >= strptr) then
    printesc(
      507)
  else
  begin
    printesc(hash[p].rh);
    printchar(32);
  end;
end;

File: cpp/src/printcs.cpp
#include "printcs.h"
#include "printesc.h"
#include "printchar.h"
#include "print.h"
#include "texte.h"

void printcs(int p)
{
	if (p < hash_base)
		if (p >= single_base)
			if (p == null_cs)
			{
				printesc("csname");
				printesc("endcsname");
				printchar(' ');
			}
			else
			{
				printesc(std::string(1, char(p-single_base)));
				if (cat_code(p-single_base) == letter)
					printchar(' ');
			}
		else 
			if (p < active_base)
				printesc("IMPOSSIBLE.");
			else
				printchar(p-active_base);
	else 
		if (p >= undefined_control_sequence)
			printesc("IMPOSSIBLE.");
		else 
			if (text(p) < 0 || text(p) >= strptr)
				printesc("NONEXISTENT.");
			else
			{
				printesc(TXT(hash[p].rh));
				printchar(' ');
			}
}

